pac4j-saml-db/src/main/java/org/pac4j/saml/dbcrypto/KeyStoreCredentialProvider2.java

```diff
@@ -0,0 +1,156 @@
+package org.pac4j.saml.dbcrypto;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.opensaml.core.criterion.EntityIdCriterion;
+import org.opensaml.security.credential.Credential;
+import org.opensaml.security.credential.CredentialResolver;
+import org.opensaml.security.credential.impl.KeyStoreCredentialResolver;
+import org.opensaml.security.x509.X509Credential;
+import org.opensaml.xmlsec.config.DefaultSecurityConfigurationBootstrap;
+import org.opensaml.xmlsec.keyinfo.KeyInfoCredentialResolver;
+import org.opensaml.xmlsec.keyinfo.KeyInfoGenerator;
+import org.opensaml.xmlsec.keyinfo.NamedKeyInfoGeneratorManager;
+import org.opensaml.xmlsec.signature.KeyInfo;
+import org.pac4j.saml.crypto.CredentialProvider;
+import org.pac4j.saml.crypto.KeyStoreCredentialProvider;
+import org.pac4j.saml.exceptions.SAMLException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.shibboleth.utilities.java.support.resolver.CriteriaSet;
+import net.shibboleth.utilities.java.support.resolver.ResolverException;
+
+
+/**
+ * Alternative to {@link KeyStoreCredentialProvider}. Unlike {@link KeyStoreCredentialProvider}, it does not take a path to the keystore but the keystore binary content.
+ *
+ * TODO: We cannot extend {@link KeyStoreCredentialProvider} because it needs a path in the constructor. It would be good to have
+ * a common ancestor to reuse some code. Most methods are just copied - duplicate code.
+ * 
+ * @author jkacer
+ */
+public class KeyStoreCredentialProvider2 implements CredentialProvider { 

```

Comment:
Done. So we have a base class (KeyStoreCredentialProvider) and two specialized sub-classes (KeyStoreFileCredentialProvider and KeyStoreByteArrayCredentialProvider). The two sub-classes only redefine the constructor, otherwise all code is inherited from the base class.